File: ABC144/ABC144_D.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int a, b, x;
    cin >> a >> b >> x;
    int v = a * a * b;
    //xが容積の半分かどうかで処理が変わる

    double ans;
    if (x <= v / 2){
        int h = 2 * x / a / b;
        ans = (atan(b / h) / M_PI * 180);
    }

    cout << ans << endl;
    return 0;
}

File: ABC144/ABC144_D.py
import math

a, b, x = map(int, input().split())
v = a * a * b
if(x <= v / 2):
    h = 2 * x / a / b
    ans = math.degrees(math.atan(b / h))
else:
    h = (v - x) / a / a * 2
    ans = math.degrees(math.atan(h / a))

print(ans)